### ISSUE TITLE
Paginated GraphQL calls use an XViewNode type as response

### ISSUE THREAD

**Issue by EugenMayer** [Blazebit/blaze-persistence#1719]
When  we declare a paginated data fetcher to return a view `CatView`, the type definitions that are generated for GraphQL will include 2 types (and more). 

- `CatView` (since this type exists as a view / java type)
- `CatViewNode` - this type is introduced by the BP GraphQL type generation

The problem is, that a paginated call will return `CatViewNodeViewConnection` in the graphqls schema, which itself nests `CatViewEdge` which then nests `CatViewNode`. 

The latter one `CatViewNode` should actually be `CatView` and those types are identical in the generated type definition, but they  are not the same.

The problem becomes more visible when the graphql client uses the schema to generate a `client-sdk` and uses types to deal with cache and flusing those. 

Is there any reason we introduced `XViewNode` in the first place? The entire type seems redundant to me.

**Comment by beikov** [comment]
See https://persistence.blazebit.com/documentation/1.6/entity-view/manual/en_US/#pagination-support for details about why this is this way. You can disable the relay schema parts by calling `setDefineRelayTypes(false)`, `setImplementRelayNode(false)` and `setDefineRelayNodeIfNotExist(false)` on the `GraphQLEntityViewSupportFactory`

**Comment by EugenMayer** [comment]
Thank you for the quick reply. I'am not sure i understand the article entirely. 
Disabling 

```java
    graphQLEntityViewSupportFactory.setImplementRelayNode(false);
    graphQLEntityViewSupportFactory.setDefineRelayNodeIfNotExist(false);
```

Will lead to 

```
core  |          * The field type 'PageInfo' is not present when resolving type 'AuditLogIdViewConnection' [@-1:-1]
core  |          * The field type 'PageInfo' is not present when resolving type 'ModerationPageModerationSettingEmbeddedIdViewConnection' [@-1:-1]
core  |          * The field type 'PageInfo' is not present when resolving type 'AttachmentExcerptViewConnection' [@-1:-1]
```

PageInfo is part of `Connection`

When using what you those options, what is the impact / burden. Does this mean we can no longer use the pagination implementation BP offers for GQL at all? Thanks for the clarification (and hopefully this will be helpful for others too)

**Comment by EugenMayer** [comment]
> You have to find a custom way to implement pagination which is not standardized.

Looking at the implementation, we could do that, seems quiet possible. But is there any technical reason you introduce an entirely new type for the node? This is not part of the spec and is not done usually https://www.apollographql.com/docs/technotes/TN0029-relay-style-connections/

Surely Connection and Edge are needed, but the inner node type could be just the EV we define as the projection, right? Is there really a technical need to introduce a new type for that?

**Comment by EugenMayer** [comment]
I would love to work on that. As soon as the BP main branch has a working GQL integration again, blocked by #1707 , we will work on this.

AFAIU we would implement

- a new flag `defineRelayNodeTypes` which is true by default 
- if enabled, XViewNode types are generated (current behavior) 
- if disabled, the actual XView type is used

Did i understand your idea correctly

**Comment by beikov** [comment]
>a new flag `defineRelayNodeTypes` which is true by default

You can make it false by default and add a note about the backwards incompatible change in the CHANGELOG

>if enabled, XViewNode types are generated (current behavior)

Correct

>if disabled, the actual XView type is used

Yes, but also make sure that `XView` implements `Node` if `isImplementRelayNode()`

**Comment by EugenMayer** [comment]
> Yes, but also make sure that XView implements Node if isImplementRelayNode() 
What does this mean? Cannot find a type /interface `Node`

i see 

```java
        if (!typeRegistry.getType("Node").isPresent() && isImplementRelayNode() && isDefineRelayNodeIfNotExist()) {
            List<FieldDefinition> nodeFields = new ArrayList<>(4);
            nodeFields.add(new FieldDefinition("id", new NonNullType(new TypeName("ID"))));
            addDefinition(typeRegistry, newInterfaceTypeDefinition("Node", nodeFields, null));
        }
```

Which basically Seem to represent

```graphqls
type Node {
  id: ID!
}
```

Right now, looking at our types, i cannot find this type nor i can see that the XViewNode type implement/extend of this. Could you explain a little on that detail?
